src/Microsoft.NetCore.Analyzers/Core/Runtime/AvoidZeroLengthArrayAllocations.cs

```diff
@@ -111,13 +111,38 @@ private static bool IsCompilerGeneratedParamsArray(IArrayCreationExpression arra
             var model = context.Compilation.GetSemanticModel(arrayCreationExpression.Syntax.SyntaxTree);
 
             // Compiler generated array creation seems to just use the syntax from the parent.
-            var parent = model.GetOperationInternal(arrayCreationExpression.Syntax, context.CancellationToken) as IInvocationExpression;
-            if (parent?.TargetMethod == null || parent.TargetMethod.Parameters.Length == 0)
+            var parent = model.GetOperationInternal(arrayCreationExpression.Syntax, context.CancellationToken) as IHasArgumentsExpression;
+            if (parent == null)
             {
                 return false;
             }
 
-            if (!parent.TargetMethod.Parameters.Last().IsParams)
+            ISymbol targetSymbol;
+            var invocation = parent as IInvocationExpression;
```

Comment:
if (parent is IInvocationExpression invocation)

(same with the rest of your as + null checks).